src/activities/Elsa.Activities.MassTransit/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -19,38 +19,43 @@ public static IServiceCollection AddRabbitMqActivities(this IServiceCollection s
         {
             var optionsBuilder = services.AddOptions<RabbitMqOptions>();
             options?.Invoke(optionsBuilder);
-            
+
             services
                 .AddActivity<SendMassTransitMessage>()
                 .AddActivity<ReceiveMassTransitMessage>();
 
             services.AddSingleton<SimplifiedBusHealthCheck>();
             services.AddSingleton<ReceiveEndpointHealthCheck>();
-            
+
             services.AddMassTransit(
                 massTransit =>
                 {
+                    foreach (var messageType in messageTypes)
+                    {
+                        massTransit.AddConsumer(CreateConsumerType(messageType));
+                    }
+
                     massTransit.AddBus(sp => CreateUsingRabbitMq(massTransit, sp, messageTypes));
                 });
 
             services.AddSingleton<IHostedService, MassTransitHostedService>();
 
-            foreach (var messageType in messageTypes)
-            {
-                var consumerType = CreateConsumerType(messageType);
-                services.AddSingleton(consumerType);
-            }
-
             return services;
         }
 
         private static IBusControl CreateUsingRabbitMq(IServiceCollectionConfigurator massTransit, IServiceProvider sp, IEnumerable<Type> messageTypes)
         {
+
             return Bus.Factory.CreateUsingRabbitMq(
                 bus =>
                 {
-                    var options = sp.GetRequiredService<IOptions<RabbitMqOptions>>();
-                    var host = bus.Host(new Uri(options.Value.Host), _ => { });
+                    var options = sp.GetRequiredService<IOptions<RabbitMqOptions>>().Value;
+                    var host = bus.Host(new Uri(options.Host), h =>
+                    {
+                        h.Username(options.Username);

```

Comment:
Would it perhaps be better if we allow the user to either include the credentials as part of the connection string, or separately? Or will this code allow for that already (and it still works if option.Username / option.Password are empty)?

Reply:
Default behaviour inside MassTransit is to use guest/guest if nothing is provided. Which is what was happening before.

Most MassTransit samples have a username password setting in the app.config, or hadcode guest/guest to be explicit about the password, so thought it would be best to do so here.

Reply:
I'm fine with that, just as long as it still works when the user uses the credentials as part of the AMQP connection string.

Reply:
I didnt even know you could set the username and password in the connection string.

I have now tested both use cases with explicit username and password, and just a host URI with username and password as defined here: https://www.rabbitmq.com/uri-spec.html